src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -124,18 +127,32 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
 
             throttle = new IndexThrottle();
             this.searcherFactory = new SearchFactory(engineConfig);
+            final Tuple<Long, Long> translogId; // nextTranslogId, currentTranslogId
             try {
                 writer = createWriter();
+                indexWriter = writer;
+                translogId = loadTranslogIds(writer, translog);
             } catch (IOException e) {
                 throw new EngineCreationFailureException(shardId, "failed to create engine", e);
             }
-            indexWriter = writer;
-            manager = createSearcherManager();
+            manager = createSearcherManager(translogId.v2());
```

Comment:
can we move the translog related code out of createSearchManager and move it here (or have a dedicated method for just that). I think it's good to have it all together.